src/array/cpu/concurrent_id_hash_map.h

```diff
@@ -24,34 +24,36 @@ namespace aten {
  * key insertions once with Init function, and it does not support key deletion.
  *
  * The hash map should be prepared in two phases before using. With the first
- * being creating the hashmap, and then init it with an id array.
+ * being creating the hashmap, and then initialize it with an id array which is
+ * divided into 2 parts: [`seed ids`, `left ids`]. In result `seed ids` are
+ * mapped to [0, num_seed_ids) and `left ids` to [num_seed_ids, num_unique_ids).
+ * Notice that mapping order is stable for `seed ids` while not for the left.
  *
- * For example, for an array A with following entries:
- * [98, 98, 100, 99, 97, 99, 101, 100, 102]
+ * For example, for an array A having 4 seed ids with following entries:
+ * [99, 98, 100, 97, 97, 101, 101, 102, 101]
```

Comment:
In the comment below you mentioned `num_seeds` ids are unique, I am assuming the first 4 are seed ids, but I see duplicated 97 in this example, it this intended?

Reply:
Yes. `Seed ids` is unique among themselves, but it can be duplicate with other ids. So put it here may help user clarify it.